zerver/data_import/import_util.py

```diff
@@ -730,8 +728,18 @@ def process_emojis(
 
         emoji_records.append(emoji_record)
 
-    # Run downloads in parallel
-    run_parallel_wrapper(partial(get_emojis, emoji_dir), upload_emoji_list, threads=threads)
+        # Directly download the emoji and patch the file_name with the correct extension
+        # based on the content-type returned by the server. This is needed because Slack
+        # sometimes returns an emoji url with .png extension despite the file being a gif.
+        content_type = get_emojis(emoji_dir, emoji_url, emoji_path)
+        if content_type is not None:
+            file_extension = guess_extension(content_type)
+            if file_extension is None:
+                continue
+            old_file_name = emoji_record["file_name"]
+            new_file_name = f"{old_file_name.rsplit('.', 1)[0]}{file_extension}"
+            emoji_record["file_name"] = new_file_name
+            emoji["file_name"] = new_file_name
```

Comment:
As explained in the commit message:
```
2. The Slack API when returning the emoji records, returns the record
for its thumbsup_all emoji with the url ending with .png, even though
the file is a gif. For that reason, we have to make that code correct
file extensions based on the response content-type.

Emojis are the smallest set of images to download, so for simplicity of
implementation, we remove the parallelization of the downloads in favor
of just processing them serially.
```

if this is making things too small, we can parallelize this i think and the parallel `get_emojis` processes can generate a combined list showing which emoji `file_name` should get corrected to what extension - which the main process can read at the end and make correction. But that makes things a bit more complicated, so i left that out in case it's worth it here. (There should only be a handful of custom emojis anyway)